simexpal/build.py

```diff
@@ -89,6 +89,17 @@ def get_concurrency():
 			n = os.cpu_count()
 		return n
 
+	def get_source_dir_for(build_name):
+		if build_name == build.name:
+			if not build.revision.is_dev_build:
+				return build.clone_dir
+			return build.source_dir
+
+		other_build = cfg.get_build(build_name, build.revision)
```

Comment:
Why do we special case the current build here?

Reply:
It saves us a call of `cfg.get_build()`, which loops over all builds and revision. I can remove this case if you think it is overkill

Reply:
That sounds more like an efficiency problem in `cfg.get_build()` to me. If this turns out to be too slow, we could always add a `dict()` of all builds.

Yes, please remove this special case here to simplify the code.

Reply:
Done.